src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -235,24 +235,29 @@ private static bool IsSelfClosing(Block beginTagBlock)
                         span.Content.Length);
                 }
 
-                attribute = default(KeyValuePair<string, SyntaxTreeNode>);
-
-                return false;
+                return null;
             }
 
+            bool isBoundNonStringAttribute;
+            var result = new TryParseResult
+            {
+                IsBoundAttribute = IsBoundAttribute(name, descriptors, out isBoundNonStringAttribute),
+                AttributeName = name,
+            };
+            result.IsBoundNonStringAttribute = isBoundNonStringAttribute;
```

Comment:
Nah.  Though setters are done in order, it's more clear to keep this "after `IsBoundAttribute`" bit separate.  I was tempted not to use constructor initialization at all but decided the mid ground was most understandable.